packages/forms/src/UIForm/fields/Datalist/displayMode/TextMode.component.js

```diff
@@ -34,9 +34,16 @@ class TextMode extends React.Component {
 
 	render() {
 		const { id, schema, value, t } = this.props;
-		const { title } = schema;
+		const { title, options } = schema;
 		const titleMap = this.state.titleMap || this.props.schema.titleMap || [];
-		const titleEntry = titleMap.find(entry => entry.value === value);
+		let titleEntry = titleMap.find(entry => entry.value === value);
+
+		if (options && options.isMultiSection) {
+			options.titleMap.find(section => {
+				titleEntry = section.suggestions.find(entry => entry.value === value);
+				return !!titleEntry;
+			});
+		}
```

Comment:
Can you wrap the non-multi-section part in an else please ?
There is no need to search the label in titleMap, then in multi-section mode, search again in option.titleMap